sips/sip-021/sip-021-trustless-two-way-peg-for-bitcoin.md

```diff
@@ -0,0 +1,1385 @@
+already-catastrophic blockchain failure (in which case, other remedies such as
+hard forks would be viable alternatives).
+
+* **The ability for miners to choose their block-commit value on-the-fly.** In
+this proposal, miners must choose a block-commit value for the next block they
+intend to mine.  However, to date miners do not appear to dynamically adjust
+their PoX expenditures every block as it is, so this feature's ongoing utility
+is questionable.
+
+# Activation
+
+This section will be filled in with the advice of the CABs.
+
+# Conclusion
+
+This proposal describes sBTC, a novel Bitcoin 2-way-peg that offers a
+PoX-powered recovery procedure for temporary peg liveness failures while
+rewarding the peg wallet's stewards (Stackers, via PoX) for staying online
+regardless of how much or little peg traffic exists.  The proposal sacrifices
+forks of arbitrary length in order to handle peg-ins and peg-outs of arbitrary
+size in a fixed amount of time in an in-band fashion, while keeping mining and
+Stacking running in an open-membership setting.  Building the peg on Stacks
+avoids complicated designs related to Bitcoin forking by ensuring that the peg
+state exists in all Bitcoin forks.  By making sure that Stacking and
+peg-operations materialize in all Stacks forks, we additionally ensure that
+mining on the canonical chain is always the most profitable course of action
+for miners.
+
+# Appendix A: New Transactions
+
+This section describes the wire formats of the various sBTC Bitcoin
+transactions.
+
+### Peg Hand-off Transaction
+
+* **Description**: This is used by the current reward cycle's Stackers to send
+the pegged-in BTC to the next reward cycle's Stackers' peg wallet address.
+
+* **Opcode**: `H`
+
+* **Data**
+
+	* **Bytes 3-11**: The reward cycle number
+(big-endian) corresponding to the Stacker set doing the hand-off
+
+* **Output 2**: the new peg wallet address
+
+* **Notes**: Anyone can send a hand-off
+transaction in order to (re-)fill the current reward cycle's peg wallet.  This
+is used to recover from loss of BTC due to transaction fees, or even wallet
+theft.
+
+
+### Peg-In Transaction
+
+* **Description**: This is used by an sBTC user to peg-in BTC and materialize
+sBTC. 
+
+* **Opcode**: `<`
+
+* **Data**
+
+	* **Bytes 3-80**: c32-encoded Stacks
+address to receive the sBTC
+
+* **Output 2**: the current peg wallet address
+
+### Peg-out Request Transaction
+
+* **Description**: This is used by an sBTC holder to request a peg-out *
+**Opcode**: `>`
+
+* **Data**
+
+	* **Bytes 3-11**: The number of sBTC to peg-out, in
+satoshis (big-endian)
+
+        * **Bytes 11-76**: A 65-byte recoverable secp256k1 low-S
+signature from the sBTC owner's private key(s), over the SHA512/256 hash of the
+bytes consisting of
+
+		* the number of sBTC (bytes 3-11 above)
+
+		* Output 2's
+scriptPubKey
+
+* **Output 2**: a dust output with the recipient Bitcoin
+scriptPubKey
+
```

Comment:
Reminder: We need an **Output 3** which sends a dust amount to the peg wallet to fund the fulfillment transaction fee, as discussed in yesterdays [sBTC meeting].(https://docs.google.com/document/d/1m4ROTYvgZhTJxbMY7NaI8N2sk5chbEm9SWEIE_Ewuy0/edit)

Related thought: If we want to be able to link fulfillment transactions and peg-out requests, should we also add a reference to the peg out request `txid` in the fulfillment transaction data? Because a single peg-out request might have multiple fulfillment transactions.